tensorflow_datasets/scripts/cli/new_dataset.py

```diff
@@ -0,0 +1,256 @@
+  # then the tests needs to provide the fake output paths relative to the
+  # fake data directory
+  # DL_EXTRACT_RESULT = {{'some_key': 'output_file1.txt', ...}}
+
+
+if __name__ == "__main__":
+  tfds.testing.test_main()
+
+"""
+
+_CHECKSUM_FILE = """\
+# {TODO}: If your dataset downloads files, then the checksums will be
+# automatically added here when running the download_and_prepare script
+# with --register_checksums.
+"""
+
+_INIT_FILE = """\
+\"""{dataset_name} dataset.\"""
+
+from {dataset_name} import {dataset_cls}
+
+__all__ = [
+    "{dataset_cls}"
+]
+"""
+
+_FAKE_DATA_GEN_FILE = """\
+from absl import app
+
+def main(_):
+  # {TODO}: Generate Fake Data
+  pass 
+
+if __name__ == "__main__":
+  app.run(main)
+"""
+
+
+def create_dataset_file(root_dir: str, data: Dict[str, str]) -> None:
+  """Create a new dataset from a template."""
+  file_path = os.path.join(root_dir, '{dataset_name}.py')
+  context = (
+      _HEADER + _DATASET_DEFAULT_IMPORTS + _CITATION + _DESCRIPTION +
+      _DATASET_DEFAULTS)
+
+  with tf.io.gfile.GFile(file_path.format(**data), 'w') as f:
+    f.write(context.format(**data))
+
+
+def add_the_init(root_dir: str, data: Dict[str, str]) -> None:
+  """Creates a new __init__.py. file"""
+  file_path = os.path.join(root_dir, '__init__.py')
+
+  with tf.io.gfile.GFile(file_path.format(**data), 'w') as f:
+    f.write(_INIT_FILE.format(**data))
+
+
+def create_dataset_test_file(root_dir: str, data: Dict[str, str]) -> None:
+  """Create the test file associated with the dataset."""
+  file_path = os.path.join(root_dir, '{dataset_name}_test.py')
+  context = (_HEADER + _DATASET_TEST_DEFAULTS_IMPORTS + _DATASET_TEST_DEFAULTS)
+
+  with tf.io.gfile.GFile(file_path.format(**data), 'w') as f:
+    f.write(context.format(**data))
+
+
+def create_fake_data(root_dir: str, data: Dict[str, str]) -> None:
+  fake_examples_dir = os.path.join(root_dir, 'fake_examples')
+  fake_examples_dir = fake_examples_dir.format(**data)
+  tf.io.gfile.makedirs(fake_examples_dir)
+
+  fake_path = os.path.join(fake_examples_dir,
+                           'TODO-add_fake_data_in_this_directory.txt')
+  with tf.io.gfile.GFile(fake_path, 'w') as f:
+    f.write('{TODO}: Add fake data in this directory'.format(**data))
+
+
+def create_fake_data_gen_file(root_dir: str, data: Dict[str, str]) -> None:
+  fake_data_generator = os.path.join(root_dir, 'fake_data_generator.py')
+
+  content = (_HEADER + _FAKE_DATA_GEN_FILE)
+  with tf.io.gfile.GFile(fake_data_generator.format(**data), 'w') as f:
+    f.write(content.format(**data))
+
+
+def create_checksum_file(root_dir: str, data: Dict[str, str]) -> None:
+  checksum_path = os.path.join(root_dir, 'checksums.txt')
+  with tf.io.gfile.GFile(checksum_path.format(**data), 'w') as f:
+    f.write(_CHECKSUM_FILE.format(**data))
+
+
+def create_single_dataset(ds_name: str) -> None:
+  root_dir = os.path.abspath(ds_name)
+  tf.io.gfile.makedirs(os.path.join(ds_name, 'fake_examples'))
+
+  data = dict(
```

Comment:
We should use dataclasses now

Reply:
Dataclasses seems to complicate the template generation as we format the string using `**data`.

Reply:
But not if we use f-string, no ?

```
file_path.write_text(textwrap.dedent(f"""\
from tensorflow.dataset.{data.name} import {data.cls_name}
"""))
```